src/blocks/packages.rs

```diff
@@ -0,0 +1,374 @@
+#[serde(rename_all = "lowercase")]
+pub enum PackageManager {
+    Apt,
+    Pacman,
+    Aur,
+    Dnf,
+}
+
+pub async fn run(config: &Config, api: &CommonApi) -> Result<()> {
+    let mut config: Config = config.clone();
+
+    let format = config.format.with_default(" $icon $total.eng(w:1) ")?;
+    let format_singular = config
+        .format_singular
+        .with_default(" $icon $total.eng(w:1) ")?;
+    let format_up_to_date = config
+        .format_up_to_date
+        .with_default(" $icon $total.eng(w:1) ")?;
+
+    // If user provide package manager in any of the formats then consider that also
+    macro_rules! any_format_contains {
+        ($name:expr) => {
+            format.contains_key($name)
+                || format_singular.contains_key($name)
+                || format_up_to_date.contains_key($name)
+        };
+    }
+
+    let apt = any_format_contains!("apt");
+    let aur = any_format_contains!("aur");
+    let pacman = any_format_contains!("pacman");
+
+    if !config.package_manager.contains(&PackageManager::Apt) && apt {
+        config.package_manager.push(PackageManager::Apt);
+    }
+    if !config.package_manager.contains(&PackageManager::Pacman) && pacman {
+        config.package_manager.push(PackageManager::Pacman);
+    }
+    if !config.package_manager.contains(&PackageManager::Aur) && aur {
+        config.package_manager.push(PackageManager::Aur);
+    }
+    if !config.package_manager.contains(&PackageManager::Dnf) && aur {
+        config.package_manager.push(PackageManager::Dnf);
+    }
+
+    let warning_updates_regex = config
+        .warning_updates_regex
+        .as_deref()
+        .map(Regex::new)
+        .transpose()
+        .error("invalid warning updates regex")?;
+    let critical_updates_regex = config
+        .critical_updates_regex
+        .as_deref()
+        .map(Regex::new)
+        .transpose()
+        .error("invalid critical updates regex")?;
+    let ignore_updates_regex = config
+        .ignore_updates_regex
+        .as_deref()
+        .map(Regex::new)
+        .transpose()
+        .error("invalid ignore updates regex")?;
+
+    let mut package_manager_vec = Vec::new();
+
+    for &package_manager in config.package_manager.iter() {
+        let backend = match package_manager {
+            PackageManager::Apt => {
+                Box::new(Apt::new(config.ignore_phased_updates).await?) as Box<dyn Backend>
+            }
+            PackageManager::Pacman => Box::new(Pacman::new().await?) as Box<dyn Backend>,
+            PackageManager::Aur => {
+                Box::new(Aur::new(config.aur_command.clone().unwrap_or_default()))
+                    as Box<dyn Backend>
+            }
+            PackageManager::Dnf => Box::new(Dnf::new()) as Box<dyn Backend>,
+        };
+
+        package_manager_vec.push(backend);
+    }
+
+    loop {
+        let mut package_manager_map: HashMap<&str, u32> =
+            [("apt", 0), ("pacman", 0), ("aur", 0), ("dnf", 0)].into();
```

Comment:
As I've written before, I would prefer to keep backend-specific code to the minimum. Start with an empty hash map and insert as you go. And to not have two hash maps (this one and the one with values to be displayed), you can insert directly into the map which you will pass to `widget.set_values(...)`).

Reply:
I did this because of below code, It may panic if "apt" doesn't exist..that's why I thought to set default 0
```rust
"apt" => Value::number(package_manager_map["apt"]),
```

Reply:
How can i get inner value from `Value::number(num)`

Reply:
Why? Here is an example what you can do https://github.com/greshake/i3status-rust/pull/1988#discussion_r1460760865

Reply:
> Here is an example what you can do https://github.com/greshake/i3status-rust/pull/1988#discussion_r1460760865

Yah, I got that but I didn't got how do i get total_count from a `HashMap<Cow<'static, str>, Value>`

Reply:
Done btw

Reply:
Why not just

```rust
package_manager_map.insert(package_manager.name(), Value::number(updates_count));
total += updates_cont;
```

Reply:
Oh sorry, I am dumb. I thought to go hard way